### ISSUE TITLE
Z-index issue when triggering modals from fixed toolbar

### ISSUE THREAD

**Issue by ciar4n** [joomla/joomla-cms#20255]
### Steps to reproduce the issue
Navigate to Components -> Smart Search
Scroll down so toolbar sticks
Click 'Index'

### Expected result
Modal appears above backdrop

### Actual result
Modal appears behind backdrop

![image](https://user-images.githubusercontent.com/2803503/39405204-ffb3ef66-4b98-11e8-804e-170410dc42dc.png)

**Comment by Quy** [comment]
It was ok in 3.9beta3 but not in 3.9beta4.

**Comment by Quy** [comment]
PR #22367 produces this issue. It was ok in beta3 but not in beta4.

**Comment by ciar4n** [comment]
That would mean it was fixed since the issue was created and broke again with #22367 ?

**Comment by Quy** [comment]
Yes that appears to be the case. I replaced the affected files in #22367 from beta3 and the issue was fixed.

**Comment by ciar4n** [comment]
The issue was gone before and after scrolling?

**Comment by Quy** [comment]
I’m not at my computer to confirm. I will check later today.

**Comment by infograf768** [comment]
My test here show that the modal is indeed dark (in fact the page itself is darkened and the modal is beneath) but the indexing is done and then the modal auto-close.

**Comment by Quy** [comment]
@infograf768 Can you test with beta3? Be sure to scroll down the page until the toolbar sticks at the top and then click the `index` button.

**Comment by infograf768** [comment]
The point is that with staging I do not need to scroll at all. The issue happens whatever the size of the window and no scroll.
Will try to get a beta3.

**Comment by ciar4n** [comment]
Please read.. https://github.com/joomla/joomla-cms/issues/22465#issuecomment-426940210

**Comment by infograf768** [comment]
beta3 => same issue!!!

**Comment by ciar4n** [comment]
Yes this issue has always been present. @Quy s original issue (#22465) is only since beta 4.

**Comment by infograf768** [comment]
Well, it has not always been present. It was working OK in 3.8.12...

**Comment by ciar4n** [comment]
No, this was always an issue. And will remain an issue even if #22367 is reverted. #22465 is a separate issue and is only present since 3.9 beta4.

**Comment by infograf768** [comment]
I disagree. I just installed a clean 3.8.12 and the modal displays fine OVER the backdrop.
(Macintosh, Firefox)
![modalissue](https://user-images.githubusercontent.com/869724/46464959-e4739700-c7c7-11e8-9218-8c12216ffb6e.gif)

**Comment by ciar4n** [comment]
And is the issue present if you scroll first then open the modal (as stated in the description)?

**Comment by infograf768** [comment]
In staging, scrolling or not scrolling is unimportant.
As I said above, the modal is hidden behind the backdrop.
The issue is unrelated to scrolling.
Tested also with scrolling in 3.8.12 and it is also unrelated. It works in both cases.

**Comment by infograf768** [comment]
I consider this as a non B/C situation. A 3rd party extension could be displaying a modal this way where informations have to be entered.

**Comment by ciar4n** [comment]
Sighs...

```
Navigate to Components -> Smart Search
Scroll down so toolbar sticks
Click 'Index'
```

This issue is everything related to scrolling. This issue was only present after scrolling. Open 3.8.12 and follow the above. I repeat...

>  #22465 is a separate issue and is only present since 3.9 beta4.

**Comment by infograf768** [comment]
In fact, you are right **when there are already indexes in the page.**
When indexes have been cleared, no issue after scrolling, as you can see below (3.8.12).


![modalissue2](https://user-images.githubusercontent.com/869724/46466452-9d87a080-c7cb-11e8-9409-4ae75bebf757.gif)

I indeed never tested Indexing when the list contains some stuff. I systematically clear Index before indexing...

EDIT: same situation in staging.
EDIT2: The difference in staging seems that it is also broken when NOT scrolling.

**Comment by infograf768** [comment]
This to show that the issue in staging happens when NOT scrolling at all and list empty.

![modalissue3](https://user-images.githubusercontent.com/869724/46467195-6dd99800-c7cd-11e8-9c5b-7ce2ecf3acad.gif)

**Comment by ciar4n** [comment]
Thank you for confirming.

Both this issue and #22465 are somewhat related in that they are both caused by anything other than `position:static` applied to the toolbar. Prior to https://github.com/joomla/joomla-cms/pull/22367 this was only after scrolling.

**Comment by infograf768** [comment]
Shall we therefore consider reverting https://github.com/joomla/joomla-cms/pull/22367 if it at least solves the issue when not scrolling, or do we have a better solution?

**Comment by ciar4n** [comment]
I think so. Unless someone can fix how modals are rendered (unlikely at this stage), the only solution, as I see it, is to revert #22367.

**Comment by ciar4n** [comment]
Note.. reverting won't fix this issue, but will fix #22465

**Comment by ciar4n** [comment]
When I say fix, rendering the modal anywhere outside of the subhead div should do it (eg. Batch modal in com-content)

**Comment by infograf768** [comment]
For finder we have
```
		$toolbar->appendButton(
			'Popup', 'archive', 'COM_FINDER_INDEX', 'index.php?option=com_finder&view=indexer&tmpl=component', 500, 210, 0, 0,
			'window.parent.location.reload()', 'COM_FINDER_HEADING_INDEXER'
		);
```

For batch, we have
```
$toolbar->popupButton('batch')
				->text('JTOOLBAR_BATCH')
				->selector('collapseModal')
				->listCheck(true);
```
Also, the issue does not only concern Indexed Content  but also the Statistics button in this view as in the other finder views (Maps & Search Filters)
```
$toolbar->appendButton('Popup', 'bars', 'COM_FINDER_STATISTICS', 'index.php?option=com_finder&view=statistics&tmpl=component', 550, 350);
```

**Comment by infograf768** [comment]
@Hackwar 
Any way to modify the code for these Toolbar buttons?

**Comment by infograf768** [comment]
Revert PR is ready here
https://github.com/joomla/joomla-cms/pull/22506

**Comment by jwaisner** [comment]
@ciar4n Can you confirm after the revert PR that this issue is fixed?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20255">issues.joomla.org/tracker/joomla-cms/20255</a>.</sub>

**Comment by Harmageddon** [comment]
@jwaisner As noted above (https://github.com/joomla/joomla-cms/issues/20255#issuecomment-426964893), the revert only fixed the issue for scenarios when you don't scroll. When scrolling down (so the toolbar becomes sticky), the issue still exists in v3.9.16.

**Comment by Hackwar** [comment]
Is this still an issue? I can't replicate this on J4 in Smart Search.

**Comment by Quy** [comment]
It is still reproducible in J3, but probably not worth fixing as it does not affect the indexing. @zero-24 close?

**Comment by Quy** [comment]
Closing as there are no more J3 releases other than security fixes. Not an issue in J4.
